inc/policyremoveapplication.class.php

```diff
@@ -90,13 +87,91 @@ public function unicityCheck($value, $itemtype, $itemId, PluginFlyvemdmFleet $fl
     * @return array|bool
     */
    public function getMqttMessage($value, $itemtype, $itemId) {
-      if (! $this->integrityCheck($value, $itemtype, $itemId)) {
+      if (!$this->integrityCheck($value, $itemtype, $itemId)) {
          return false;
       }
 
+      // force the itemtype for packages
+      if (!($itemtype instanceof PluginFlyvemdmPackage)) {
+         $itemtype = new PluginFlyvemdmPackage();
+      }
+
+      $itemtype->getFromDB($itemId);
       $array = [
-            $this->symbol  => $value
+         $this->symbol  => $this->findPackageName($itemtype),
+         'id'           => $itemtype->getID(),
       ];
       return $array;
    }
+
+   /**
+    * @param PluginFlyvemdmFleet $fleet
+    * @param mixed $value
+    * @param mixed $itemtype
+    * @param int $itemId
+    * @return bool
+    */
+   public function apply(PluginFlyvemdmFleet $fleet, $value, $itemtype, $itemId) {
+      $this->canApply = false;
+
+      // force the itemtype for packages
+      if (!($itemtype instanceof PluginFlyvemdmPackage)) {
+         $itemtype = new PluginFlyvemdmPackage();
+      }
```

Comment:
same as above. itemtype is a string (a type), not an instance

Reply:
as we discussed today, this if() block becomes useless, there is no need to save an itemtype or a items_id on the DB for app removal.